superset-frontend/src/dashboard/util/getFilterScopeFromNodesTree.js

```diff
@@ -22,11 +22,29 @@ import { flatMap, isEmpty } from 'lodash';
 import { CHART_TYPE, TAB_TYPE } from './componentTypes';
 import { getChartIdAndColumnFromFilterKey } from './getDashboardFilterKey';
 
+function getChartsFromTabsNotInScope({ tabs = [], tabsInScope = [] }) {
+  const chartsNotInScope = [];
+  tabs.forEach(({ value: tab, children: tabChildren }) => {
+    if (tabChildren && !tabsInScope.includes(tab)) {
+      tabChildren.forEach(({ value: subTab, children: subTabChildren }) => {
+        if (subTabChildren && !tabsInScope.includes(subTab)) {
+          chartsNotInScope.push(
+            ...subTabChildren.filter(({ type }) => type === CHART_TYPE),
+          );
+        }
+      });
+    }
+  });
+
+  // return chartId only
+  return chartsNotInScope.map(({ value }) => value);
```

Comment:
I feel like this could probably be done more succinctly with a `reduce`? I can try to work my way through it as a suggestion if you'd like

Reply:
I would use `reduce` a few month ago...Recently @ktmud gave me a code review comment and says `reduce` is hard to read (for many ppl)...like this thread:
https://twitter.com/jaffathecake/status/1213077702300852224?lang=en

So now i try to avoid using `reduce`.

Reply:
lol TIL!

I do wonder what would happen if we all used `reduce` more often; maybe it would be as easy to read as `map`! But this works :D